engine/runtime/src/main/scala/org/enso/compiler/Compiler.scala

```diff
@@ -297,6 +297,7 @@ class Compiler(
     var hasInvalidModuleRelink = false
     if (irCachingEnabled) {
       requiredModules.foreach { module =>
+        ensureParsed(module)
```

Comment:
Why is this line added here? Could you pls add a comment? In the above code, I can see that under certain circumstances, there is `parseModule(module)` called. Isn't that enough?

Reply:
I also don't understand this change. Required module should already be parsed.

Reply:
Removing the `parseModule(module)` here leads to:
```
[error] Test org.enso.compiler.SerdeCompilerTest.testFibTest failed: java.lang.NullPointerException: Cannot invoke "org.enso.compiler.core.ir.Module.preorder()" because the return value of "org.enso.compiler.context.CompilerContext.getIr(org.enso.compiler.context.CompilerContext$Module)" is null, took 0.257 sec
[error]     at org.enso.compiler.Compiler.$anonfun$runCompilerPipeline$6(Compiler.scala:303)
[error]     at org.enso.compiler.Compiler.$anonfun$runCompilerPipeline$6$adapted(Compiler.scala:299)
[error]     at scala.collection.immutable.List.foreach(List.scala:333)
[error]     at org.enso.compiler.Compiler.runCompilerPipeline(Compiler.scala:299)
[error]     at org.enso.compiler.Compiler.go$1(Compiler.scala:233)
[error]     at org.enso.compiler.Compiler.runInternal(Compiler.scala:240)
[error]     at org.enso.compiler.Compiler.run(Compiler.scala:136)
[error]     at org.enso.compiler.SerdeCompilerTest.parseSerializedModule(SerdeCompilerTest.java:89)
[error]     at org.enso.compiler.SerdeCompilerTest.testFibTest(SerdeCompilerTest.java:36)
```
why?

Now, when we use `BindingsMap` in the `runImportsAndExportsResolution`, it may happen that modules get to this phase without having `IR` (just the `BindingsMap`). However the `requiredModules` will need `IR` to pass thru `runGlobalTypingPasses` at line 350. E.g. we have to load the `IR` for them. For example at this moment.